Description: Only use the first word of GRUB_DISTRIBUTOR for --class
Origin: upstream, http://bazaar.launchpad.net/~vcs-imports/grub/grub2-bzr/revision/2296
Bug-Ubuntu: https://bugs.launchpad.net/bugs/557606
Forwarded: yes
Last-Update: 2010-04-08

diff -Nur -x '*.orig' -x '*~' grub2/util/grub.d/10_hurd.in grub2.new/util/grub.d/10_hurd.in
--- grub2/util/grub.d/10_hurd.in	2010-03-09 16:14:00.000000000 +0000
+++ grub2.new/util/grub.d/10_hurd.in	2010-04-08 11:04:31.000000000 +0100
@@ -27,7 +27,7 @@
   OS=GNU
 else
   OS="${GRUB_DISTRIBUTOR} GNU/Hurd"
-  CLASS="--class $(echo ${GRUB_DISTRIBUTOR} | tr '[A-Z]' '[a-z]') ${CLASS}"
+  CLASS="--class $(echo ${GRUB_DISTRIBUTOR} | tr '[A-Z]' '[a-z]' | cut -d' ' -f1) ${CLASS}"
 fi
 
 at_least_one=false
diff -Nur -x '*.orig' -x '*~' grub2/util/grub.d/10_kfreebsd.in grub2.new/util/grub.d/10_kfreebsd.in
--- grub2/util/grub.d/10_kfreebsd.in	2010-03-09 16:14:00.000000000 +0000
+++ grub2.new/util/grub.d/10_kfreebsd.in	2010-04-08 11:04:31.000000000 +0100
@@ -30,7 +30,7 @@
 case "${GRUB_DISTRIBUTOR}" in
   Debian)
 	OS="${GRUB_DISTRIBUTOR} GNU/kFreeBSD"
-	CLASS="--class $(echo ${GRUB_DISTRIBUTOR} | tr '[A-Z]' '[a-z]') --class gnu-kfreebsd --class gnu ${CLASS}"
+	CLASS="--class $(echo ${GRUB_DISTRIBUTOR} | tr '[A-Z]' '[a-z]' | cut -d' ' -f1) --class gnu-kfreebsd --class gnu ${CLASS}"
   ;;
   *)
 	OS="FreeBSD"
diff -Nur -x '*.orig' -x '*~' grub2/util/grub.d/10_linux.in grub2.new/util/grub.d/10_linux.in
--- grub2/util/grub.d/10_linux.in	2010-04-08 11:00:30.000000000 +0100
+++ grub2.new/util/grub.d/10_linux.in	2010-04-08 11:04:49.000000000 +0100
@@ -31,7 +31,7 @@
   OS=GNU/Linux
 else
   OS="${GRUB_DISTRIBUTOR}"
-  CLASS="--class $(echo ${GRUB_DISTRIBUTOR} | tr '[A-Z]' '[a-z]') ${CLASS}"
+  CLASS="--class $(echo ${GRUB_DISTRIBUTOR} | tr '[A-Z]' '[a-z]' | cut -d' ' -f1) ${CLASS}"
 fi
 
 # loop-AES arranges things so that /dev/loop/X can be our root device, but
